flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityConfiguration.java

```diff
@@ -77,7 +87,8 @@ public SecurityConfiguration(Configuration flinkConf) {
 	 * @param securityModuleFactories the security modules to apply.
 	 */
 	public SecurityConfiguration(Configuration flinkConf,
```

Comment:
@aljoscha I am not sure whether this answers your question: the default security factories (module and context) is used when user doesn't explicitly set them. If users want to use their own - they will have to use this API to explicitly set the context and all module factories

Reply:
How would they do that? I think this is only possible if you setup your own TaskManagerRunners and other runners, for example for YARN. Is this the intended purpose?